salt/pillar/__init__.py

```diff
@@ -1219,8 +1228,11 @@ def compile_pillar(self, ext=True):
         top, top_errors = self.get_top()
         if ext:
             if self.opts.get("ext_pillar_first", False):
-                self.opts["pillar"], errors = self.ext_pillar(self.pillar_override)
-                self.rend = salt.loader.render(self.opts, self.functions)
+                pillar, errors = self.ext_pillar(self.pillar_override)
```

Comment:
Without this, the `tests.unit.test_pillar.PillarTestCase.test_ext_pillar_first` test fails for me because the matchers are looking for `pillar` in `opts` where it is not present without this:
https://github.com/saltstack/salt/blob/master/salt/matchers/pillar_match.py#L30
fails with `UnboundLocalError: cannot access local variable 'pillar' where it is not associated with a value`
```suggestion
                # matchers needs pillar in opts
                self.opts["pillar"] = self.pillar_data
                pillar, errors = self.ext_pillar(self.pillar_override)
```

Reply:
^ Patching the test doesn't work, but my suggestion above does

Reply:
Maybe pillar isn't populated due to the changes Pedro suggested above